packages/block-library/src/cover/transforms.js

```diff
@@ -52,6 +55,42 @@ const transforms = {
 					]
 				),
 		},
+		{
+			type: 'block',
+			blocks: [ 'core/group' ],
+			isMatch: ( { backgroundColor, gradient, style } ) => {
+				/*
+				 * Make this transformation available only if the Group has background
+				 * or gradient set, because otherwise `Cover` block displays a Placeholder.
+				 *
+				 * This helps avoid arbitrary decisions about the Cover block's background
+				 * and user confusion about the existence of previous content.
+				 */
+				return (
+					backgroundColor ||
+					style?.color?.background ||
+					style?.color?.gradient ||
+					gradient
+				);
+			},
+			transform: (
+				{ align, anchor, backgroundColor, gradient, style },
+				innerBlocks
+			) => {
+				return createBlock(
+					'core/cover',
+					{
+						align,
+						anchor,
+						overlayColor: backgroundColor,
+						customOverlayColor: style?.color?.background,
+						gradient,
+						customGradient: style?.color?.gradient,
+					},
+					createBlocksFromInnerBlocksTemplate( innerBlocks )
```

Comment:
Hey - following on my previous comment here: https://github.com/WordPress/gutenberg/pull/30890#discussion_r630316027, this seems that is not needed at all here. Passing just the `innerBlocks` is enough.
```suggestion
				innerBlocks
```

`Group` and this transform are different in some things and are not handled in the same way. This issue is not related to your PR and will be investigated a bit more if there is some problem in the reducers' logic.

Reply:
Okay! Thanks much for all the work looking into it.

This should be updated now.